### PR TITLE
fix(huobi) - fetchOHLCV

### PR DESCRIPTION
**Pull request by ttodua**
I have seen this PR : https://github.com/ccxt/ccxt/pull/18014
but it is not quite correct. for example, check this query, this works for 1month and 1year too:

https://api.huobi.pro/market/history/candles?period=1mon&size=200&symbol=btcusdt&from=1621642000&to=1681643000

the previously mentioned query didn't work because it's `from` time was `1291132800` which is too early,  before the symbol listing time, so it throws issue, but if you place `from` in correct range, it works for monthly and yearly too. so, i think we should do this, when `from <> to` are not too far from each other, all timeframes, even 1-year tf works

(i also didnt use `method` variable, because it's fragile for userland codes if/when we update the implicit api methods, which is frequent for huobi, so better to be somewhat generic like "useHistoricalEndpoint")

```
npm run cli.js htx fetchOHLCV BTC/USDT 1M 1620000000000   

> ccxt@4.2.68 cli.js
> node ./examples/js/cli.js htx fetchOHLCV BTC/USDT 1M 1620000000000

2024-03-12T17:33:05.742Z
Node.js: v20.10.0
CCXT v4.2.68
htx.fetchOHLCV (BTC/USDT, 1M, 1620000000000)
2024-03-12T17:33:08.379Z iteration 0 passed in 454 ms

1622476800000 | 36961.31 |    41328 |    28800 | 34198.93 | 1005585.6934037823
1625068800000 |  34195.5 |    42400 | 29278.45 | 41540.32 |  516358.5581759969
1627747200000 | 41540.32 |    50500 | 37305.38 | 47398.95 |  507050.2982356468
```

### CODE REVIEW

File: ts/src/htx.ts [hunk 1]
```diff
@@ -3025,18 +3028,21 @@ export default class htx extends Exchange {
...
-            }
-            if (limit !== undefined) {
-                request['size'] = limit; // max 2000
-            }
             request['symbol'] = market['id'];
-            if (timeframe === '1M' || timeframe === '1y') {
-                // for some reason 1M and 1Y does not work with the regular endpoint
-                // https://github.com/ccxt/ccxt/issues/18006
+            let useHistorical = undefined;
+            [ useHistorical, params ] = this.handleOptionAndParams (params, 'fetchOHLCV', 'useHistoricalEndpoint', true);
```
Comment (carlosmiei) [inline ts/src/htx.ts:3033]: we're missing documenting this param in the js docs

File: ts/src/htx.ts [hunk 2]
```diff
@@ -951,6 +951,9 @@ export default class htx extends Exchange {
                         },
                     },
                 },
+                'fetchOHLCV': {
+                    'useHistoricalEndpoint': true,
```
[the remark this answers is not in the document]
Comment (ttodua) [reply ts/src/htx.ts:955]: We already used this endpoint, the only exclusion was 1year/1month.
the only problem that arose there was for user who requested data from too past and from/to disatance was abnormal.
so, I think that by default we could use this endpoint, as long as we use this endpoint for all other timeframes (1m...) already

**Comment by ttodua** [comment]
@carlosmiei I think PR would pass, but unrelated issue pops up.

**Comment by ttodua** [comment]
@carlosmiei can you help me to sort out this PR? something weird issue happens, I have not touched those parts.

**Comment by ttodua** [comment]
@carlosmiei PR passed eventually.

**Pull request merged**